spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/LogisticRegressionTrainer.java

```diff
@@ -0,0 +1,162 @@
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Spark Sink plugin that trains a model based upon various labels in the structured record.
+ * Writes this model to a file of a FileSet.
+ */
+@Plugin(type = SparkSink.PLUGIN_TYPE)
+@Name(LogisticRegressionTrainer.PLUGIN_NAME)
+@Description("Uses Logistic Regression to train a model based upon whether messages are spam or not.")
+public class LogisticRegressionTrainer extends SparkSink<StructuredRecord> {
+  public static final String PLUGIN_NAME = "LogisticRegressionTrainer";
+
+  private Config config;
+
+  /**
+   * Configuration for the LogisticRegressionTrainer.
+   */
+  public static class Config extends PluginConfig {
+
+    @Description("The name of the FileSet to save the model to.")
+    private String fileSetName;
+
+    @Description("Path of the FileSet to save the model to.")
+    private String path;
+
+    @Nullable
+    @Description("A comma-separated sequence of fields that needs to be used for training.")
+    private String featureFieldsToInclude;
+
+    @Nullable
+    @Description("A comma-separated sequence of fields that needs to be excluded from being used in training.")
+    private String featureFieldsToExclude;
+
+    @Description("The field from which to get the label. It must be of type double.")
+    private String labelField;
+
+    @Nullable
+    @Description("The number of features to use in training the model. It must be of type integer and equal to the" +
+      " number of features used in LogisticRegressionClassifier. The default value if none is provided will be" +
+      " 100.")
+    private final Integer numFeatures;
+
+    @Nullable
+    @Description("The number of classes to use in training the model. It must be of type integer. " +
+      "The default value if none is provided will be 2.")
+    private final Integer numClasses;
+
+    public Config() {
+      this.numFeatures = 100;
+      this.numClasses = 2;
+    }
+
+    public Config(String fileSetName, String path, @Nullable String featureFieldsToInclude, String labelField,
+                  @Nullable String featureFieldsToExclude, @Nullable Integer numFeatures,
+                  @Nullable Integer numClasses) {
+      this.fileSetName = fileSetName;
+      this.path = path;
+      this.featureFieldsToInclude = featureFieldsToInclude;
+      this.featureFieldsToExclude = featureFieldsToExclude;
+      this.labelField = labelField;
+      this.numFeatures = numFeatures;
+      this.numClasses = numClasses;
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    pipelineConfigurer.createDataset(config.fileSetName, FileSet.class);
+
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    Preconditions.checkArgument(inputSchema != null, "Input Schema must be a known constant.");
+    SparkUtils.validateConfigParameters(inputSchema, config.featureFieldsToInclude, config.featureFieldsToExclude,
```

Comment:
as mentioned above, this would be cleaner if it was in a concrete parent class instead of a utils class.